weld/subsystem/src/main/java/org/jboss/as/weld/deployment/processors/ExternalBeanArchiveProcessor.java

```diff
@@ -162,6 +162,14 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
                 return;
             }
             for (DependencySpec dep : module.getDependencies()) {
+                if (!(dep instanceof ModuleDependencySpec)) {
+                    continue;
+                }
+                if (deploymentUnits.stream().anyMatch(d -> (((ModuleDependencySpec) dep).getName()).endsWith(d.getName()))) {
+                    // d.getName() returns deployment.123abc.ear, whereas dep.getName() returns 123abc.ear
```

Comment:
This comment is confusing because for using the example values the code would amount to "123abc.ear".endsWith("deployment.123abc.ear") which would never be true.

Reply:
Oh yea, it's the other way around of course, my bad. Either way, I wasn't planning to keep it here, it was just a temporary explanation of why the matching works based on my debugging...